test fails in a container due to new ordering of checks

With eoan as a host (kernel 5.x), on a trusty container, a few tests fail at package build time:
```
================================== FAILURES ===================================
 TestLivepatchUserFacingStatus.test_user_facing_status_inapplicable_on_inapplicable_status 

self = <uaclient.entitlements.tests.test_livepatch.TestLivepatchUserFacingStatus object at 0x14fb74653908>
entitlement = <uaclient.entitlements.livepatch.LivepatchEntitlement object at 0x14fb750104a8>

    def test_user_facing_status_inapplicable_on_inapplicable_status(
        self, entitlement
    ):
        """The user-facing details INAPPLICABLE applicability_status"""
        livepatch_bionic = entitlement.cfg.read_cache(
            "machine-access-livepatch"
        )
        livepatch_bionic["entitlement"]["affordances"]["series"] = ["bionic"]
        entitlement.cfg.write_cache(
            "machine-access-livepatch", livepatch_bionic
        )
    
        with mock.patch("uaclient.util.get_platform_info") as m_platform_info:
            m_platform_info.return_value = PLATFORM_INFO_SUPPORTED
            uf_status, details = entitlement.user_facing_status()
        assert uf_status == status.UserFacingStatus.INAPPLICABLE
        expected_details = (
            "Livepatch is not available for Ubuntu 16.04 LTS"
            " (Xenial Xerus)."
        )
>       assert expected_details == details
E       assert 'Livepatch is...enial Xerus).' == 'Cannot instal...n a container'
E         - Livepatch is not available for Ubuntu 16.04 LTS (Xenial Xerus).
E         + Cannot install Livepatch on a container

uaclient/entitlements/tests/test_livepatch.py:81: AssertionError
 TestLivepatchEntitlementCanEnable.test_can_enable_false_on_unsupported_kernel_min_version 

self = <uaclient.entitlements.tests.test_livepatch.TestLivepatchEntitlementCanEnable object at 0x14fb74650ba8>
capsys = <_pytest.capture.CaptureFixture object at 0x14fb7465cba8>
entitlement = <uaclient.entitlements.livepatch.LivepatchEntitlement object at 0x14fb7464a048>

    def test_can_enable_false_on_unsupported_kernel_min_version(
        self, capsys, entitlement
    ):
        """"False when on a kernel less or equal to minKernelVersion."""
        unsupported_min_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
        unsupported_min_kernel["kernel"] = "4.2.9-00-generic"
        with mock.patch("uaclient.util.get_platform_info") as m_platform:
            m_platform.return_value = unsupported_min_kernel
            entitlement = LivepatchEntitlement(entitlement.cfg)
            assert not entitlement.can_enable()
        msg = (
            "Livepatch is not available for kernel 4.2.9-00-generic.\n"
            "Minimum kernel version required: 4.3\n"
        )
>       assert (msg, "") == capsys.readouterr()
E       assert ('Livepatch i...d: 4.3\n', '') == ('Cannot insta...tainer\n', '')
E         At index 0 diff: 'Livepatch is not available for kernel 4.2.9-00-generic.\nMinimum kernel version required: 4.3\n' != 'Cannot install Livepatch on a container\n'

uaclient/entitlements/tests/test_livepatch.py:187: AssertionError
 TestLivepatchEntitlementCanEnable.test_can_enable_false_on_unsupported_kernel_flavor 

self = <uaclient.entitlements.tests.test_livepatch.TestLivepatchEntitlementCanEnable object at 0x14fb7464fa90>
capsys = <_pytest.capture.CaptureFixture object at 0x14fb7464fc88>
entitlement = <uaclient.entitlements.livepatch.LivepatchEntitlement object at 0x14fb74666dd8>

    def test_can_enable_false_on_unsupported_kernel_flavor(
        self, capsys, entitlement
    ):
        """"When on an unsupported kernel, can_enable returns False."""
        unsupported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
        unsupported_kernel["kernel"] = "4.4.0-140-notgeneric"
        with mock.patch("uaclient.util.get_platform_info") as m_platform:
            m_platform.return_value = unsupported_kernel
            entitlement = LivepatchEntitlement(entitlement.cfg)
            assert not entitlement.can_enable()
        msg = (
            "Livepatch is not available for kernel 4.4.0-140-notgeneric.\n"
            "Supported flavors are: generic, lowlatency\n"
        )
>       assert (msg, "") == capsys.readouterr()
E       assert ('Livepatch i...atency\n', '') == ('Cannot insta...tainer\n', '')
E         At index 0 diff: 'Livepatch is not available for kernel 4.4.0-140-notgeneric.\nSupported flavors are: generic, lowlatency\n' != 'Cannot install Livepatch on a container\n'

uaclient/entitlements/tests/test_livepatch.py:203: AssertionError
 TestLivepatchEntitlementCanEnable.test_can_enable_false_on_unsupported_architecture 

self = <uaclient.entitlements.tests.test_livepatch.TestLivepatchEntitlementCanEnable object at 0x14fb746725c0>
capsys = <_pytest.capture.CaptureFixture object at 0x14fb74672588>
entitlement = <uaclient.entitlements.livepatch.LivepatchEntitlement object at 0x14fb74dd91d0>

    def test_can_enable_false_on_unsupported_architecture(
        self, capsys, entitlement
    ):
        """"When on an unsupported architecture, can_enable returns False."""
        unsupported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
        unsupported_kernel["arch"] = "ppc64le"
        with mock.patch("uaclient.util.get_platform_info") as m_platform:
            m_platform.return_value = unsupported_kernel
            assert not entitlement.can_enable()
        msg = (
            "Livepatch is not available for platform ppc64le.\n"
            "Supported platforms are: x86_64\n"
        )
>       assert (msg, "") == capsys.readouterr()
E       assert ('Livepatch i...x86_64\n', '') == ('Cannot insta...tainer\n', '')
E         At index 0 diff: 'Livepatch is not available for platform ppc64le.\nSupported platforms are: x86_64\n' != 'Cannot install Livepatch on a container\n'

uaclient/entitlements/tests/test_livepatch.py:218: AssertionError
==================== 4 failed, 463 passed in 2.17 seconds =====================
```

--- a/uaclient/entitlements/tests/test_livepatch.py
+++ b/uaclient/entitlements/tests/test_livepatch.py
@@ -58,8 +58,11 @@ class TestLivepatchContractStatus:
 
 
 class TestLivepatchUserFacingStatus:
+    @mock.patch(
+        "uaclient.entitlements.livepatch.util.is_container", return_value=False
+    )
     def test_user_facing_status_inapplicable_on_inapplicable_status(
-        self, entitlement
+        self, _m_is_container, entitlement
     ):
         """The user-facing details INAPPLICABLE applicability_status"""
         livepatch_bionic = entitlement.cfg.read_cache(
@@ -157,13 +160,16 @@ class TestLivepatchProcessConfigDirectives:
         assert 0 == m_subp.call_count
 
 
+@mock.patch(
+    "uaclient.entitlements.livepatch.util.is_container", return_value=False
+)
 class TestLivepatchEntitlementCanEnable:
     @pytest.mark.parametrize(
         "supported_kernel_ver",
         ("4.4.0-00-generic", "5.0.0-00-generic", "4.19.0-00-generic"),
     )
     def test_can_enable_true_on_entitlement_inactive(
-        self, supported_kernel_ver, capsys, entitlement
+        self, _m_is_container, supported_kernel_ver, capsys, entitlement
     ):
         """When entitlement is INACTIVE, can_enable returns True."""
         supported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
@@ -177,7 +183,7 @@ class TestLivepatchEntitlementCanEnable:
         assert [mock.call()] == m_container.call_args_list
 
     def test_can_enable_false_on_unsupported_kernel_min_version(
-        self, capsys, entitlement
+        self, _m_is_container, capsys, entitlement
     ):
         """"False when on a kernel less or equal to minKernelVersion."""
         unsupported_min_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
@@ -193,7 +199,7 @@ class TestLivepatchEntitlementCanEnable:
         assert (msg, "") == capsys.readouterr()
 
     def test_can_enable_false_on_unsupported_kernel_flavor(
-        self, capsys, entitlement
+        self, _m_is_container, capsys, entitlement
     ):
         """"When on an unsupported kernel, can_enable returns False."""
         unsupported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
@@ -219,7 +225,12 @@ class TestLivepatchEntitlementCanEnable:
         ),
     )
     def test_can_enable_false_on_unsupported_min_kernel_version(
-        self, kernel_version, meets_min_version, capsys, entitlement
+        self,
+        _m_is_container,
+        kernel_version,
+        meets_min_version,
+        capsys,
+        entitlement,
     ):
         """"When on an unsupported kernel version, can_enable returns False."""
         unsupported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
@@ -241,7 +252,7 @@ class TestLivepatchEntitlementCanEnable:
         assert (msg, "") == capsys.readouterr()
 
     def test_can_enable_false_on_unsupported_architecture(
-        self, capsys, entitlement
+        self, _m_is_container, capsys, entitlement
     ):
         """"When on an unsupported architecture, can_enable returns False."""
         unsupported_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
@@ -255,16 +266,17 @@ class TestLivepatchEntitlementCanEnable:
         )
         assert (msg, "") == capsys.readouterr()
 
-    def test_can_enable_false_on_containers(self, capsys, entitlement):
+    def test_can_enable_false_on_containers(
+        self, m_is_container, capsys, entitlement
+    ):
         """When is_container is True, can_enable returns False."""
         unsupported_min_kernel = copy.deepcopy(dict(PLATFORM_INFO_SUPPORTED))
         unsupported_min_kernel["kernel"] = "4.2.9-00-generic"
         with mock.patch("uaclient.util.get_platform_info") as m_platform:
-            with mock.patch("uaclient.util.is_container") as m_container:
-                m_platform.return_value = unsupported_min_kernel
-                m_container.return_value = True
-                entitlement = LivepatchEntitlement(entitlement.cfg)
-                assert not entitlement.can_enable()
+            m_platform.return_value = unsupported_min_kernel
+            m_is_container.return_value = True
+            entitlement = LivepatchEntitlement(entitlement.cfg)
+            assert not entitlement.can_enable()
         msg = "Cannot install Livepatch on a container\n"
         assert (msg, "") == capsys.readouterr()
 

